test/utils/asserts.hpp

```diff
@@ -23,13 +25,57 @@ MATCHER(Result, "empty result")
 
 MATCHER_P(Result, value, "")  // NOLINT(readability-redundant-string-init)
 {
+    using namespace fizzy;
+
+    // Require the arg to be TypedExecutionResult.
+    // This can be a static_assert, but just returning false and failing a test provides better
+    // location of the error.
+    using result_type = std::remove_cv_t<std::remove_reference_t<arg_type>>;
+    if constexpr (!std::is_same_v<result_type, test::TypedExecutionResult>)
+    {
+        if (result_listener->IsInterested())
```

Comment:
Where is this part of google.test documented?

Reply:
I found this in implementation of builtin matchers. This seems not to be strictly needed, but this is false e.q. when `Not()` is used.

Reply:
Seems to be undocumented feature. The only mention of `result_listener` I found uses only `operator<<` https://github.com/google/googletest/blob/master/googlemock/docs/cheat_sheet.md#defining-matchers